### PR TITLE
Gb/tmy files

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

File: rex/resource_extraction/resource_extraction.py [hunk 1]
```diff
@@ -971,6 +1008,44 @@ def get_timestep_map(self, ds_name, timestep, region=None,
...
+            with cls(res_h5) as res:
+                res.get_SAM_gid(gids, out_path=out_path,
+                                write_time=write_time, **kwargs)
+        else:
+            msg = 'Bad gids dtype: {}'.format(type(gids))
+            assert isinstance(gids, (list, tuple, np.ndarray)), msg
+            gid_chunks = np.array_split(np.array(gids), n_chunks)
+            with SpawnProcessPool(max_workers=max_workers) as spp:
+                for chunk in gid_chunks:
+                    spp.submit(cls.make_SAM_files, res_h5, chunk, out_path,
```
Comment (MRossol) [inline rex/resource_extraction/resource_extraction.py:1046]: Beautiful recursion!

File: rex/resource_extraction/wind_cli.py [hunk 2]
```diff
@@ -110,10 +110,14 @@ def sam_datasets(ctx, hub_height, lat_lon, gid, sites):
         logger.info('Saving data to {}'.format(ctx.obj['OUT_DIR']))
         with ctx.obj['CLS'](ctx.obj['H5'], **ctx.obj['CLS_KWARGS']) as f:
             if lat_lon is not None:
+                out_path = os.path.join(ctx.obj['OUT_DIR'],
+                                        'SAM_{}.csv'.format(lat_lon))
                 f.get_SAM_lat_lon(hub_height, lat_lon,
-                                  out_path=ctx.obj['OUT_DIR'])
+                                  out_path=out_path)
```
Comment (MRossol) [inline rex/resource_extraction/wind_cli.py:116]: Was this a bug? OR does this need to be updated in the other CLIs?

**Review by MRossol — APPROVED** [review]
One question but looks awesome. Nice clean up of my code and great recursion!

File: rex/resource_extraction/wind_cli.py (hunk 2 above)
Comment (grantbuster) [reply rex/resource_extraction/wind_cli.py:116]: yeah you know im not 100% sure how that worked! I added an assert statement in the `get_SAM_lat_lon()` and `get_SAM_gid()` methods that checked the out path ended in .csv and it failed here!

**Pull request merged**